### PR TITLE
Alberta Caregiver Amount

### PR DESCRIPTION
**Pull request by Kenyaokun**
First try #331

Co-authored-by: YiweiJ <YiweiJ@users.noreply.github.com>

### CODE REVIEW

**Comment by Kenyaokun** [comment]
I actually run several times make format, but the Lint would not pass, not sure why it does not work @MaxGhenis @PavelMakarchuk

**Comment by MaxGhenis** [comment]
Try `pip install -U black`?

**Comment by Kenyaokun** [comment]
Cool, it works!

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/credits_return/lower_dependant_income_threshold.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+description: Alberta Lower Income threshold for dependants.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/credits_return/lower_dependant_income_threshold.yaml:1]:
```suggestion
description: Alberta provides the caregiver amount for dependants with income below this threshold.
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/credits_return/lower_age_eligibility.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+description: Alberta limits its Child and Family Benefit to people above this age.
+values:
+  2023-01-01: 18
+metadata:
+  unit: year
+  label: Alberta child and Family Benefit child age eligibility
+  reference:
+  - title: Province of Alberta - Alberta Child and Family Benefit
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1ab/td1ab-23e.pdf
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/credits_return/lower_age_eligibility.yaml:9]: could you add the legal code reference to the parameter files (the alberta law)

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/credits_return/lower_age_eligibility.yaml:9]: all yaml

File: policyengine_canada/variables/household/person/is_parent_or_grandparent.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+
+class is_parent_or_grandparent(Variable):
+    value_type = bool
+    entity = Person
+    label = "Is parent or grandparent"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        return person("age", period) >= 65
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/household/person/is_parent_or_grandparent.py:11]: make 65 a parameter and this a blank class (seems to be Alberta specific)

File: policyengine_canada/variables/gov/provinces/ab/tax/credits_return/ab_caregiver_amount.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        )
+        dependant_net_income = (
+            person("individual_net_income", period) * is_dependant
+        )
+
+        income_eligibility = (
+            dependant_net_income <= p.upper_dependant_income_threshold
+        )
+
+        caregiver_amount = (
```
[the remark this answers is not in the document]
Comment (Kenyaokun) [reply policyengine_canada/variables/gov/provinces/ab/tax/credits_return/ab_caregiver_amount.py:31]: We don't need the amount for the eligible dependants: [https://www.canada.ca/en/revenue-agency/services/forms-publications/tax-packages-years/general-income-tax-benefit-package/alberta/5009-pc/information-residents-alberta.html](https://www.canada.ca/en/revenue-agency/services/forms-publications/tax-packages-years/general-income-tax-benefit-package/alberta/5009-pc/information-residents-alberta.html) The website mentioend that "If you and another person support the same dependant, you can split the claim for that dependant. However, the total amount of your claim and the other person's claim cannot be more than the maximum amount allowed for that dependant." I think why we should include the maxmium amount.

File: changelog_entry.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Alberta Caregiver Amount.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Alberta caregiver amount.
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/credits_return/age_eligibility.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+description: Alberta limits its Child and Family Benefit to people above this age.
+values:
+  2023-01-01: 18
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/credits_return/age_eligibility.yaml:3]: find 2022 values

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/credits_return/age_eligibility.yaml:3]: all yaml files

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/credits_return/age_eligibility.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+metadata:
+  unit: year
+  label: Alberta child and Family Benefit adult dependant age eligibility
+  reference:
+  - title: Province of Alberta - Alberta Child and Family Benefit
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1ab/td1ab-23e.pdf
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/credits_return/age_eligibility.yaml:9]: add pdf pages

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/credits_return/age_eligibility.yaml:9]: all yaml files

File: policyengine_canada/variables/gov/provinces/ab/tax/credits_return/live_together.py [hunk 8]
```diff
@@ -0,0 +1,11 @@
+from policyengine_canada.model_api import *
+
+
+class live_together(Variable):
+    value_type = bool
+    entity = Person
+    label = "Albert Caregiver Dependant"
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/credits_return/live_together.py:7]:
```suggestion
    label = "Dependant living with the head of household"
```

File: policyengine_canada/variables/gov/provinces/ab/tax/credits_return/live_together.py [hunk 9]
```diff
@@ -0,0 +1,11 @@
+from policyengine_canada.model_api import *
+
+
+class live_together(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/credits_return/live_together.py:4]:
```suggestion
class cohabitating_dependant(Variable):
```

File: policyengine_canada/variables/gov/provinces/ab/tax/credits_return/live_together.py [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+    value_type = bool
+    entity = Person
+    label = "Albert Caregiver Dependant"
+    documentation = "Whthere your spouse's or common-law partner's dependant lives with you or not."
+    definition_period = YEAR
+    reference = "chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1ab/td1ab-23e.pdf"
+    defined_for = ProvinceCode.AB
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/credits_return/live_together.py:11]:
```suggestion
```

File: policyengine_canada/variables/gov/provinces/ab/tax/credits_return/live_together.py
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/credits_return/live_together.py:1]: move this to the household folder

**Comment by PavelMakarchuk** [comment]
Duplicate of #487

**Pull request closed without merging**